src/rgb/contract/assignments.rs

```diff
@@ -59,18 +59,19 @@ impl AssignmentsVariant {
             blinding_factors.push(amount::BlindingFactor::new(&SECP256K1_ZKP, &mut rng));
         }
 
-        // We need the last factor to be equal to the sum
-        let blinding_inputs: Vec<_> = inputs.iter().map(|inp| inp.blinding.clone()).collect();
-        if !blinding_inputs.is_empty() {
-            blinding_factors.pop();
-
-            let blinding_correction = SECP256K1_ZKP
-                .blind_sum(blinding_inputs.clone(), blinding_factors.clone())
-                .expect("SECP256K1_ZKP failure has negligible probability");
-            blinding_factors.push(blinding_correction.clone());
+        // We need the last factor to be equal to the difference
+        let mut blinding_inputs: Vec<_> = inputs.iter().map(|inp| inp.blinding.clone()).collect();
+        if blinding_inputs.is_empty() {
+            blinding_inputs.push(secp256k1zkp::key::ONE_KEY);
```

Comment:
Here and below: why `ONE_KEY`, not `ZERO_KEY`?

Reply:
because i think `ZERO_KEY` is not a valid private key and causes the InvalidSecretKey error sometimes, i believe when there are no other inputs and that's the only blinding factor.

The way I understood this is that the value used as reference is pretty much arbitrary, it just has to be one fixed value that everybody agrees on, but to be honest I'm not a ZK expert or anything, so this should be reviewed by somebody who knows these stuff.

Bottom line is, if you change that to `ZERO_KEY` it crashes, and it doesn't with `ONE_KEY`.

Reply:
Very strange. We will ask Andrew Poelstra to review this as was planned (after test coverage & completion of all code docs), so I will specifically address why this happens.

Also, not happy with the fact that we have to use Grin forked lib for bulletproofs; and will ask him to backport bulletproofs to normal lib.